### PR TITLE
kratos: Simple password checker

### PR DESCRIPTION
**Pull request by hugotiburtino**

### CODE REVIEW

**Review by elbotho — CHANGES_REQUESTED** [review]
Hej @hugotiburtino,
thanks for the PR. I have some requests, should I comment them all or take over this feature?

**Comment by hugotiburtino** [comment]
> should I comment them all or take over this feature?

You can take it over, thanks ;)

**Comment by elbotho** [comment]
@hugotiburtino do you agree with my changes?

- files should be named like their export
- more generic rules to make it work better for german passwords as well
- results in shortened list

File: src/helper/check-for-simple-password.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export function checkForSimplePassword(password?: string) {
+  if (!password) return false
+  if (/^[0-9]+$/.test(password)) return true
```
Comment (hugotiburtino) [inline src/helper/check-for-simple-password.ts]:
```suggestion
```
A very long number is secure enough

File: src/data/en/index.ts [hunk 2]
```diff
@@ -345,6 +345,7 @@ export const instanceData = {
       register: {
         registerTitle: 'Register your Serlo Account',
         passwordRequirements: 'At least 8 characters, longer is better.',
+        passwordTooEasy: 'This password is quite easy to guess. Make it longer or mix in letters, numbers and special characters.',
```
Comment (hugotiburtino) [inline src/data/en/index.ts]: Actually, mixing character types is not recommended.

> Don't require mixtures of characters types

[https://www.ory.sh/docs/concepts/password-policy#best-practices](https://www.google.com/url?q=https://www.ory.sh/docs/concepts/password-policy%23best-practices&sa=D&source=docs&ust=1670505703220758&usg=AOvVaw2CO5j2rCo-p-XWBXvYERy2)


Here are the reasons:
image

File: src/helper/check-for-simple-password.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  if (password.length < 12) {
+    const withoutCommonNumbers = password
+      .replace('123', '')
+      .replace('1234', '')
+      .replace('1', '')
+    if (/^[a-z]+$/.test(withoutCommonNumbers)) return true
```
Comment (hugotiburtino) [inline src/helper/check-for-simple-password.ts:9]: Actually, it can be hard enough:
image

**Review by hugotiburtino — COMMENTED** [review]
@elbotho 
About the refactorings sure thanks 👍
Regarding rules: to be honest I don't think we can make it so dynamic without using a library.

**Pull request closed without merging**